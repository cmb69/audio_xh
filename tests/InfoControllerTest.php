<?php

namespace Audio;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $sut = new InfoController(
            "./plugins/audio/",
            new FakeSystemChecker(),
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["audio"])
        );
        $response = $sut();
        $this->assertSame("Audio 1.1-dev", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
