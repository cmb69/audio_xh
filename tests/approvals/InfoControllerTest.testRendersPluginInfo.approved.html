
<h1>Audio 1.0</h1>
<div class="audio_syscheck">
  <h2>System check</h2>
  <p class="xh_fail">PHP version ≥ 7.1.0</p>
  <p class="xh_fail">CMSimple_XH version ≥ 1.7.0</p>
  <p class="xh_fail">Plib_XH version ≥ 1.3</p>
  <p class="xh_warning">Folder './plugins/audio/css/' writable</p>
  <p class="xh_warning">Folder './plugins/audio/languages/' writable</p>
</div>
